File: AppPy/model/NaivOnArray.py
#Algoritmo basado en el libro METHODS OF MATRIX MULTIPLICATION AN OVERVIEW OF SEVERAL METHODS AND THEIR IMPLEMENTATION y apoyado en IA

def NaivOnArrayImpl(matrix_A, matrix_B):
    #Obtiene las cantidades de columnas y filas   
    rows_A = len(matrix_A)
    cols_A = len(matrix_A[0])
    cols_B = len(matrix_B[0])

    #Inicializa con ceros la matriz result
    result = [[0.0] * cols_B for _ in range(rows_A)]  # Crear matriz de resultados

    # Realiza la multiplicación de matrices
    for i in range(rows_A):
        for j in range(cols_B):
            result[i][j] = 0.0  
            for k in range(cols_A):
                # Realiza la multiplicación de elementos y suma los resultados
                result[i][j] += matrix_A[i][k] * matrix_B[k][j]

    return result

File: AppCs/AppCs/Algoritmos/NaivOnArray.cs
//Algoritmo basado en el libro METHODS OF MATRIX MULTIPLICATION AN OVERVIEW OF SEVERAL METHODS AND THEIR IMPLEMENTATION y apoyado en IA
using System;
using services.interfaces;

class NaivOnArray : AlgorithmInterface {
    public long[][] MultiplyMatrices(long[][] matrix1, long[][] matrix2)
    {
        return naivOnArray(matrix1,matrix2);
    }

    /// <summary>
    /// La multiplicación se realiza fila por columna. 
    /// Los parámetros de entrada son las matrices A y B, y el tamaño de las matrices junto con el tamaño del resultado (N, P, M). 
    /// El resultado de la multiplicación se almacena en la matriz Result.
    /// </summary>
    /// <param name="A">Matriz A.</param>
    /// <param name="B">Matriz B.</param>
    /// <returns>Matriz donde se almacenará el resultado.</returns>
    public long[][] naivOnArray(long[][] A, long[][] B)
    {
        //Obtiene las cantidades de columnas y filas   
        int N = A.Length;
        int P = B[0].Length;
        int M = B.Length;
        //Inicializa con 0 la matriz resultado
        long[][] result = new long[N][];
        for (int o = 0; o < N; o++)
        {
            result[o] = new long[M];
        }
        // Realiza la multiplicación de matrices
        for (int i = 0; i < N; i++)
        {
            result[i] = new long[M];
            for (int j = 0; j < M; j++)
            {
                result[i][j] = 0;
                for (int k = 0; k < P; k++)
                {   
                    //Realiza la multiplicación de elementos y suma los resultados
                    result[i][j] += A[i][k] * B[k][j];
                }
            }
        }

        return result;
    }

}
